exec/java-exec/src/main/java/org/apache/drill/exec/service/ServiceEngine.java

```diff
@@ -82,10 +89,32 @@ public Controller getController() {
     return controller;
   }
 
+  private void submit(ExecutorService p, final String name, final Closeable c) {
```

Comment:
I agree about removing closeQuietly but this sounds like a separate PR.
This one is already bigger than it should, so I'm trying not to do too much refactoring in it.
I'm happy to do the closeQuietly refactoring as a follow up.